File: Contests/alcoholic.py
bottle
alcohol
n = input()
x = input()

sum = 0.0

for i in range(int(n)):
    bottle= input()
    alcohol = input()
    tmp = int(bottle)*int(alcohol)/100.0
    sum+=tmp
    if sum>x:
        print(i+1)
        return 0

print(-1)

File: Contests/alcoholic.cpp
#include<iostream>

using namespace std;

int main(){
    int n;
    int x;
    int bottels[10010];
    int alcohol[10010];
    long double amount = 0;
    cin >> n >> x;
    for(int i = 1; i <= n; i++){
        cin >> bottels[i] >> alcohol[i];
        long long int tmp = bottels[i]*alcohol[i];
        long double sum = tmp/100;
        //cout << sum << endl;
        amount+=sum;
        //cout << amount << endl;
        if(amount>x){
            cout << i << endl;
            return 0;
        }
    }

    cout << -1 << endl;

    return 0;
}